docs/tableconfig_schema.rst

```diff
@@ -0,0 +1,172 @@
+..
+.. Licensed to the Apache Software Foundation (ASF) under one
+.. or more contributor license agreements.  See the NOTICE file
+.. distributed with this work for additional information
+.. regarding copyright ownership.  The ASF licenses this file
+.. to you under the Apache License, Version 2.0 (the
+.. "License"); you may not use this file except in compliance
+.. with the License.  You may obtain a copy of the License at
+..
+..   http://www.apache.org/licenses/LICENSE-2.0
+..
+.. Unless required by applicable law or agreed to in writing,
+.. software distributed under the License is distributed on an
+.. "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+.. KIND, either express or implied.  See the License for the
+.. specific language governing permissions and limitations
+.. under the License.
+..
+
+Table Config
+=======================
+
+Table Config
+-------------
+
+Introduction to table configs
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Using tables is how Pinot serves and organizes data. There are many settings in the table config which will influence how Pinot operates. The first and most significant distinction is using an offline versus a realtime table.
```

Comment:
Consider rewording.
Some of this seems unnecessary? There should probably be a section long before this discussing the data model of offline and realtime tables.

Reply:
One concern I have here is that I don't want to go straight into offline/realtime tables without giving some preface of what we are discussing. 

My thought is if I start it such as the following, I would expect a common question/thought to be "what is offline/realtime and how are they different"?

> In this section we will describe how to configure offline and realtime tables. 

There is a section in Introduction > Architecture > Pinot Tables, but this is a long description and not very succinct. I simply wanted to give a 1-2 sentence intro to set the stage for what's be discussed. 

I am not sure which route to go here to be honest.

Reply:
Here's some options that come to mind at the moment

1. Keep the sentences I have and link to the section I mentioned above. 
2. Remove my descriptions here and give a succinct 1-2 sentence introduction to what an offline table and what a realtime table is in the section I mentioned above and then simply get straight to business here with a link to the detailed section.

Reply:
Up to you. I'd try to present a consistent format throughout your documentation though.

For example, rather than giving the paragraph style, you could include some information in your description of the possible options for `tableType` below.

Reply:
Reformatted and fixed.